Some cosmetics + additional checks on s&s

anegrin opened the pull request:
fixing some typos, introducing some checks, removing some files that should have not been commited, adding .gitignore

1. adding a comment on SMART_DISABLE_START_STOP (about C1)
2. adding a comment on SHOW_PARAMS_ON_DASHBOARD about master/slave and fixing typo
3. adding a definitions check: if SHOW_PARAMS_ON_DASHBOARD is defined then SMART_DISABLE_START_STOP should not (my understanding is they work on different cans)
4. adding .gitignroe for files/dir that should not be commited as we're using STM32CubeIDE (see [this](https://community.st.com/t5/stm32cubeide-mcus/standard-gitignore-project-setup-for-sharing-stm32cube-projects/m-p/653022/highlight/true#M25393) and [this](https://www.reddit.com/r/embedded/comments/scezb8/comment/hu88swv/))
5. removing files that shouldn't be commited: IDE configs and Release/Debug folders

anegrin commented:
@gaucho1978 some mods from my local fork (while waiting for a couple of boards :) )

gaucho1978 commented:
After deleting that files, did you ask for recompile? Are they automatically regenerated? I'm not confident in deleting files not under my control and I don't want to corrupt the project

anegrin commented:
yes so this is what you can do for testing:

```
shell cmds:
git clone --no-checkout https://github.com/anegrin/BACCAble.git
cd BACCAble
git checkout wip

then from IDE
Close current baccable project (right mouse button -> Close Project)
Delete the project (DO NOT select Delete proejct content, this will just delete the project from IDE, not from disk)
"Open Projects from File System..."
Import source: c:\temp\BACCable\firmware (example of folder with wip branch)
select firmware\ledsStripController only
press Finish
Project -> Build All
inspecting wip branch folder you'll see Debug and .settings folder properly generated

then you can go back to previous local project:

Close wip baccable project (right mouse button -> Close Project)
Delete the project (you can delete files if you like)
"Open Projects from File System..."
Import your original source folder using "Open Projects from File System..."

```

please note i've updated the PR to:

- keep ld files (they are generate byt the ide but seems find to keep them)
- deleted SavvyCAN-Windows_x64 folder and provided .url file with link for download

if you accept this PR please do "Squash and Merge"

anegrin commented:
> Please join telegram group

joined

anegrin commented:
ok so i've improved the .gitignore and removed some files from the remote repo (won't be deleted in your local one); .gitignore contains:

```
# STM32CubeIDE
.metadata <- not needed IDE will generate it if needed on project import
.settings <- not needed IDE will generate it if needed on project import
.launch <- not neeed, every dev can have its own set of launch files

# outputs
Debug <- output for binary firmware, IDE will generate it
Release <- output for binary firmware, IDE will generate it
SavvyCAN-Windows_x64 <- toll downloadble from github, provided url file

# temp files
~* <- temp files shound not be commited
```

I've also zipped UCAN step file so we go from 12Mb to 2Mb

if tested this branch by checking it out in clean folder and then importing and building with IDE

```
git clone --no-checkout https://github.com/anegrin/BACCAble.git
cd BACCAble
git checkout wip
```

and i get elf files generated correctly for both Debug and Release
